Libraries/Text/TextInput/RCTBackedTextInputDelegateAdapter.m

```diff
@@ -217,9 +221,11 @@ - (BOOL)control:(NSControl *)control textView:(NSTextView *)fieldEditor doComman
     //escape
   } else if (commandSelector == @selector(cancelOperation:)) {
     [textInputDelegate textInputDidCancel];
-    [[_backedTextInputView window] makeFirstResponder:nil];
+    if (!hasValidKeys) {
```

Comment:
I am not quite following. Can you elaborate?

Reply:
The issue is that we can't access the keyCode at this place > so we can't simple check for ESCAPE key being pressed.

Even if we could > in case you want to intercept ESCAPE in validsKeyDown or validsKeyUp we would still want to avoid calling
```[_backedTextInputView.window makeFirstResponder:nil];```

Reply:
We talked about the offline, but I thought about this more. If we made it to line 223, then we _know_ that "Escape" was pressed. So we don't need to check the keycode. So could we do something like this?

```objective c
  } else if (commandSelector == @selector(cancelOperation:)) {
       if (![validKeysDown containsObject:@"Escape"]) {
         [textInputDelegate textInputDidCancel];
         [_backedTextInputView.window makeFirstResponder:nil];
         commandHandled = YES;
       } 
       ...
  }
```

Also, would we not need to do similar checks for the other commands listed above?

Reply:
Right now we only need it for the Escape case as we are calling
```
     [_backedTextInputView.window makeFirstResponder:nil];
```
otherwise.

I have update the sample and also simplified the code based on your recommendation.
Thanks!